util/src/main/java/com/powsybl/openrao/util/InterTemporalPool.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package com.powsybl.openrao.util;
+
+import com.powsybl.openrao.commons.OpenRaoException;
+import com.powsybl.openrao.commons.TemporalData;
+import com.powsybl.openrao.commons.TemporalDataImpl;
+import org.apache.commons.lang3.tuple.Pair;
+
+import java.time.OffsetDateTime;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ForkJoinPool;
+import java.util.concurrent.ForkJoinTask;
+import java.util.function.Function;
+
+/**
+ * @author Thomas Bouquet {@literal <thomas.bouquet at rte-france.com>}
+ */
+public class InterTemporalPool extends ForkJoinPool {
```

Comment:
May need to check what happens if each task has their own children subthreads. (This class might think the task is no longer being run while the task is waiting for its subthreads to finish, and start running new tasks. I'm not 100% sure how fork join pools work in details.)